packages/storybook/src/utilities/matrix.ts

```diff
@@ -110,16 +110,23 @@ export function createMatrixInteractionsFromStates<
     THover extends readonly unknown[],
     THoverActive extends readonly unknown[],
     TActive extends readonly unknown[],
-    TFocus extends readonly unknown[]
+    TFocus extends readonly unknown[],
+    TVisited extends readonly unknown[],
+    TVisitedCombinations extends {
+        plain: TVisited[],
+        active: TActive[],
+        focus: TFocus[]
+    }
 >(
     component: (
-        ...states: THover | TActive | THoverActive | TFocus
+        ...states: THover | TActive | THoverActive | TFocus | TVisited
     ) => ViewTemplate,
     states: {
         hover: THover[],
         hoverActive: THoverActive[],
         active: TActive[],
-        focus: TFocus[]
+        focus: TFocus[],
+        visited?: TVisitedCombinations
```

Comment:
Can you follow the existing pattern and flatten these out instead of making nested options objects?

```
visited?:
visitedActive?:
visitedFocus?:
```